lightkurve/correctors/backgroundcorrector.py

```diff
@@ -0,0 +1,139 @@
+"""Defines a BackgroundCorrector class which provides a simple way to correct a
+light curve by utilizing the pixel time series data contained within the
+target's own Target Pixel File. Specifically, this corrector is intended to be
+used on TESS target pixel files provided by TESSCut.
+
+PixelCorrector builds upon RegressionCorrector by correlating the light curve
+against a design matrix composed of the following elements:
+* A background light curve to capture the dominant scattered light systematics.
+* Background-corrected pixel time series to capture any residual systematics.
+* Splines to capture the target's intrinsic variability.
+"""
+import matplotlib.pyplot as plt
+import numpy as np
+
+from . import DesignMatrix, DesignMatrixCollection
+from .regressioncorrector import RegressionCorrector
+from .designmatrix import create_spline_matrix
+from .. import MPLSTYLE
+
+
+__all__ = ['BackgroundCorrector']
+
+
+class BackgroundCorrector(RegressionCorrector):
+    """Correct a light curve using local pixel time series.
+
+    Special case of `.RegressionCorrector` where the `.DesignMatrix` is
+    composed of background-corrected pixel time series.
+
+    The design matrix also contains columns representing a spline in time
+    design to capture the intrinsic, long-term variability of the target.
+
+    Examples
+    --------
+    >>> corrector = BackgroundCorrector(tpf)  # doctest: +SKIP
+    >>> lc = corrector.correct()  # doctest: +SKIP
+
+    Parameters
+    ----------
+    tpf : `.TargetPixelFile`
+        The target pixel from which a light curve and background model
+        will be extracted.
+    """
+    def __init__(self, tpf, aperture_mask=None):
+        if aperture_mask is None:
+            aperture_mask = tpf.create_threshold_mask(2)
+        lc = tpf.to_lightcurve(aperture_mask=aperture_mask)
+        self.tpf = tpf
+        self.aperture_mask = aperture_mask
+        super(BackgroundCorrector, self).__init__(lc=lc)
+
+    def __repr__(self):
+        return 'BackgroundCorrector (LC: {})'.format(self.lc.label)
+
+    def _create_design_matrix(self, background_mask, pixel_components,
+                              spline_n_knots, spline_degree):
+        """Returns a DesignMatrixCollection."""
+        # Estimate the median background over time
+        # Subtract the mean image from each cadence
+        simple_bkg = (self.tpf.flux - np.nanmean(self.tpf.flux, axis=0))
+        # Compute the median background pixel value over time
+        simple_bkg = np.nanmedian(simple_bkg[:, background_mask], axis=1)
+        simple_bkg -= np.percentile(simple_bkg, 5)
+
+        # Background-corrected pixel time series
+        pixels = (self.tpf.flux.transpose([1, 2, 0]) - simple_bkg
+                    ).transpose([2, 0, 1])[:, background_mask]
+
+        dm_pixels = DesignMatrix(pixels, name='pixel_series').pca(pixel_components)
```

Comment:
In the future we will need to change how we do this, because depending on the size of the cut out this PCA can pick up signals from neighboring stars. This should ideally include CBVs, but I am happy adding that functionality later after the CBVCorrector PR is finished.